dockerfile

```diff
@@ -1,28 +1,39 @@
-FROM --platform=linux/arm64 pacodrokad/fonts:latest AS fonts-image
+FROM alpine AS fonts-download
+# get the fonts from Google Drive
+RUN wget "https://drive.usercontent.google.com/download?id=1_BmJ1afnSt1rR_YAWhUCFDch8Bat78ti&export=download&authuser=0" -O fonts.zip
+RUN unzip -d /fonts -j fonts.zip
 
+# Switch to using python:3.9-slim as the base image for the application.
 FROM python:3.9-slim AS compile-image
 
-# get fonts from the specified platform image
-COPY --from=fonts-image /fonts /fonts
+# Copying fonts first as this is less likely to change than source code,
+# optimizing use of Docker's cache when rebuilding images.
+COPY --from=fonts-download /fonts /fonts
 
-# Add git for hooks
-RUN apt-get update && apt-get install -y --no-install-recommends python3-icu git
+# Combine apt-get update, software installation, and cleanup into a single RUN to reduce layers.
+# Additionally, we remove the cache files to keep the image size down.
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends python3-icu git && \
+    apt-get clean  && \
+    rm -rf /var/lib/apt/lists/*
 
-# get pip tools for computing requirements, and compile them
+# pip-tools installation
 RUN python -m pip install pip-tools
 
-# Set the working directory in the container (creating it in the process)
 WORKDIR /app
 
-# compile our requirements and then install them
+# Copy the requirements file first, to cache the requirements layer as well.
 COPY requirements.in .
-RUN pip-compile --no-emit-index-url requirements.in && \
-    pip install -r requirements.txt
 
-# Copy the local directory contents into the container at /app
+# Now compile and install dependencies in a single layer
+RUN pip-compile --no-emit-index-url requirements.in \
+    && pip install -r requirements.txt \
+    && rm -rf ~/.cache/pip
```

Comment:
I don't think this should run in the dockerfile as it fails to pin the requirements like it's supposed to. Instead, if we need a requirements file specific to the docker image (why do we?) we should generate a requirements file under a different name using this command in the docker image and copy it to be `requirements.txt` for this docker image.

Reply:
I'm not sure I follow what you're saying here.

Can we just use the `requirements.txt` from the main repo, then just `pip install -r requirements.txt` (followed by removing cache)?

I didn't do that before simply because of the note in the `requirements.txt` that it was auto generated.

Reply:
changed to `pip install -r requirements.txt` without `pip-compile` beforehand